lib/classes/status/class-migrations.php

```diff
@@ -0,0 +1,241 @@
+      }
+    }
+  }
+
+  /**
+   * Returns the id of the migration that is currently running (if any)
+   * 
+   * @param array $state
+   * @return string|bool|null
+   */
+  private function _set_running_migration_id($state) {
+    $this->running_migration_id = '';
+
+    if ( !empty($state) ) {
+      if ( isset($state['is_migration']) && $state['is_migration'] ) {
+        $this->running_migration_id = $state['id']; // migration is running
+        
+        return;
+      }
+    }
+  }
+
+  /**
+   * Generate the message for the migration, depending on the current state
+   * 
+   * @param string $id
+   * @param array $migration
+   * @return string
+   */
+  private function _get_migration_message($id, $migration) {
+    $message = $migration['message'] ?? '';
+
+    switch ($migration['status']) {
+      case Migrator::STATUS_FINISHED:
+        $format = get_option('date_format') . ' ' . get_option( 'time_format' );
+        $date = wp_date($format, $migration['finished']);
+        $message = sprintf( __('Finished at %s', ud_get_stateless_media()->domain), $date );
+        break;
+      case Migrator::STATUS_SKIPPED:
+        $message = !empty($message) ? $message : __('Not required', ud_get_stateless_media()->domain);
+        break;
+      case Migrator::STATUS_FAILED:
+        $message = sprintf( __('Failed to finish: %s', ud_get_stateless_media()->domain), $message );
+        break;
+      case Migrator::STATUS_PENDING:
+        if ( $id == $this->primary_migration_id ) {
+          $message = sprintf( __('Ready to run', ud_get_stateless_media()->domain) );
+        } else {
+          $message = sprintf( 
+            __('Requires <strong>%s</strong> to finish', ud_get_stateless_media()->domain), 
+            $this->migrations[$this->primary_migration_id]['description'] 
+          );
+        }
+        break;
+      case Migrator::STATUS_RUNNING:
+        $message = __('In progress...', ud_get_stateless_media()->domain);
+        break;
+    }
+
+    return $message;
+  }
+
+  /**
+   * Returns the status of the migration
+   * 
+   * @param string $status
+   * @return string
+   */
+  public static function get_status_text($status) {
+    switch ($status) {
+      case Migrator::STATUS_PENDING:
+        return __('Pending', ud_get_stateless_media()->domain);
+      case Migrator::STATUS_RUNNING:
+        return __('Running', ud_get_stateless_media()->domain);
+      case Migrator::STATUS_SKIPPED:
+        return __('Skipped', ud_get_stateless_media()->domain);
+      case Migrator::STATUS_FINISHED:
+        return __('Finished', ud_get_stateless_media()->domain);
+      case Migrator::STATUS_FAILED:
+        return __('Failed', ud_get_stateless_media()->domain);
+    }
+  }
+
+  /**
+   * Get migrations state for frontend display
+   * 
+   * @return array
+   */
+  private function _get_migrations_state($state = null) {
+    $migrations = [];
+    $this->migrations = get_site_option(Migrator::MIGRATIONS_KEY, []);
```

Comment:
@balexey88 let's ensure that is an `array` to prevent fatal errors if it is a `string`

Reply:
Done https://github.com/udx/wp-stateless/commit/e93b7550742f3e31acc6e30449340e30b70588b0